dradis/dradis-burp
# Importing Burp Html reports

### Spec
When uploading Burp XML files into Dradis, all of the request/response data is included. 
But in Html files, Burp only shows the relevant portion of the request/response data in their HTML reports , and the rest is hidden under `[SNIP]` portions.

This PR supports parsing Burp Html files so we can see those `snipped` request/response fields. 

### How to test

- Without using the the new plugin branch, upload a burp XML file. Then switch to the new branch and upload it again in another project (in the new branch, in the plugins dropdown you should see 2 different entries for burp::xml and burp::html). Compare both projects to check they look the same. 

- Upload the XML again using the command line in another project. The upload should still look the same (`thor dradis:plugins:burp:upload /path/to/burp_xml.xml`)

- Using Burp obtain a xml report and a html file for the same burp project.
   Upload the files in 2 different projects using the web UX: (same burp output exported as XML and  HTML):
Both projects should look exactly the same, with l few minor known differences: 
    - In the HTML one there is no IP in the Evidence, just the host name.
    - In the HTML one there is no Location field in the Evidence.
    - In the XML there is a "length" truncation error. In the HTML project that error is to present because the "request/response" fields have [SNIP] sections

- Using the command line upload a burp html file in a new project, and assert the project looks exactly the same as when we upload it using the web UX. (`PROJECT_ID=N thor dradis:plugins:burp:upload /path/to/burp_html.html`)
